service/timeline.go

```diff
@@ -14,9 +14,14 @@ import (
 
 func (uis *UIServer) timelineJson(w http.ResponseWriter, r *http.Request) {
 	projCtx := MustHaveProjectContext(r)
+	project, err := projCtx.GetProject()
```

Comment:
it might be viable to have a MustHaveProject() method that panics to avoid overly verbose error handling.

Reply:
I prefer this, because it means you have explicitly pass the `project` variable.